### ISSUE TITLE
Error: Error: Permission denied, cannot open /dev/ttyAMA0

### ISSUE THREAD

**Issue by eciramella** [cncjs/cncjs#273]
#### Description

After following all the steps in the FAQ, I'm still hitting this error.

I'm wondering if I've done something wrong or misunderstood how this operates.

The desire it to connect the shapeoko 3 via USB to a Raspberry PI.

Also - it's a bit confusing, this link:

https://cnc.js.org/docs/rpi-setup-guide/#install-install-nodejs-via-node-version-manager-nvm

Suggests installing as root, while this link:

https://github.com/cncjs/cncjs

does not.

#### Versions

- CNCjs: 1.9.13
- Node.js: 6.12.3
- NPM: 5.6.0

#### How Do You Install CNCjs?

- [x] NPM
- [ ] Download the CNCjs Desktop Application

#### CNC Controller

- [x] Grbl
- [ ] Smoothieware
- [ ] TinyG/g2core

#### Hardware

- [x] Raspberry Pi
- [ ] Desktop or Laptop
- [ ] Mobile Device

#### Operating System

- [x] Not Applicable
- [ ] Windows
- [ ] Mac
- [ ] Linux

**Comment by AustinSaintAubin** [comment]
/dev/ttyAMA0 is a built in serial connection on the Rasp Pi. 
But is [must be enabled](https://learn.adafruit.com/adafruits-raspberry-pi-lesson-5-using-a-console-cable/enabling-serial-console) if you want to use it... but you dont want to use this, you want to use your USB serial connection.

You should see and use your USB serial connection ( /dev/ttyUSB0 ).

<img width="345" alt="screen shot 2018-02-08 at 12 07 20 am" src="https://user-images.githubusercontent.com/969780/35958029-29657510-0c64-11e8-9bb6-d72a736d93c8.png">

**Comment by eciramella** [comment]
Hmm - I never see ttyUSB0, I ended up using/seeing ttyACM (as per this discussion, seems to be ok http://community.carbide3d.com/t/shapeoko-3-xxl-raspberry-pi-3-cncjs-tablet/2458/25).

If this all makes sense, and OK, then we can probably close this ticket.

If I should be using /dev/ttyUSB0, I'm wondering if we can't still figure this out and update the documentation together?

One final thought, I made up a project with Carbide Create, just a basic 2" circle that gets cut .5" deep.  When I cut it with Carbide Motion, it's smooth and fast.  When I cut it with CNCjs, it's herky/jerky and much slower.  Is this POTENTIALLY because I'm not using /dev/ttyUSB0?

I can smooth it out a bit by adjusting these settings (at the substantial cost of speed):

![screen shot 2018-02-10 at 1 20 14 pm](https://user-images.githubusercontent.com/1242035/36065692-2fa9b404-0e6c-11e8-8177-e229ffc7a27a.png)

But I'm not sure what those things even are at this point (Feed rate, speed, and R?).

Also - thank you so much for the feedback, I'd love a shot a taking all the lessons learned and improving your documentation for the next person.

**Comment by eciramella** [comment]
I see this in syslog:

![screen shot 2018-02-10 at 5 01 20 pm](https://user-images.githubusercontent.com/1242035/36067086-17cd0cec-0e84-11e8-92a4-ebeddca1d3d9.png)

**Comment by eciramella** [comment]
Welp - I think we can close this ticket.

I hooked up a Dell10v (yeah yeah, me and my commodity hardware...) and it runs just fine.  No stuttering, no stumbling.

The original raspberry pi I was using was a Model B, version 1 (https://en.wikipedia.org/wiki/Raspberry_Pi).  I even overclocked it up to 900mhz and still saw the stuttering.  The issue was the planner buffer was getting saturated (I could see it get full in the grbl widget).  As I understand it, that means that the Shapeoko Grbl board is waiting for instructions.  I understand it's just waiting a few milliseconds, but it's stuttering none the less.

The Raspberry Pi 3's have a 64 bit quad core 1.2 Ghz processor.  That's 2x the power of this old Dell 10v (roughly).

Again, the goal is to create a dust proof CNC setup so I'm off to find one of those bad boys.

Closing...
